aptos-move/aptos-vm-types/src/abstract_write_op.rs

```diff
@@ -94,18 +94,39 @@ impl AbstractResourceWriteOp {
         &self,
         state_key: &StateKey,
         executor_view: &dyn ExecutorView,
+        fix_prev_materialized_size: bool,
     ) -> PartialVMResult<u64> {
         use AbstractResourceWriteOp::*;
-        match self {
-            Write(_)
-            | WriteWithDelayedFields(WriteWithDelayedFieldsOp { .. })
-            | InPlaceDelayedFieldChange(_)
-            | ResourceGroupInPlaceDelayedFieldChange(_) => Ok(executor_view
-                .get_resource_state_value_size(state_key)?
-                .unwrap_or(0)),
-            WriteResourceGroup(GroupWrite {
-                prev_group_size, ..
-            }) => Ok(*prev_group_size),
+        if fix_prev_materialized_size {
+            match self {
+                Write(_) | WriteWithDelayedFields(_) => Ok(executor_view
+                    .get_resource_state_value_size(state_key)?
+                    .unwrap_or(0)),
+                InPlaceDelayedFieldChange(InPlaceDelayedFieldChangeOp {
```

Comment:
we could still merge this, right, the idea being they are both in place (delayed field change), so materialized size works.